be/src/exec/vectorized/hdfs_scanner_text.cpp

```diff
@@ -149,17 +164,32 @@ Status HdfsTextScanner::_parse_csv(int chunk_size, ChunkPtr* chunk) {
         }
 
         bool has_error = false;
-        for (int j = 0, k = 0; j < _scanner_params.materialize_slots.size(); j++) {
+        for (int j = 0; j < _scanner_params.materialize_slots.size(); j++) {
+            int index = _scanner_params.materialize_index_in_chunk[j];
             const Slice& field = fields[_scanner_params.materialize_slots[j]->id() - 1];
             options.type_desc = &(_scanner_params.materialize_slots[j]->type());
-            if (!_converters[k]->read_string(_column_raw_ptrs[k], field, options)) {
+            if (!_converters[j]->read_string(_column_raw_ptrs[index], field, options)) {
                 chunk->get()->set_num_rows(num_rows);
                 has_error = true;
                 break;
             }
-            k++;
         }
         num_rows += !has_error;
+        if (!has_error) {
+            for (int p = 0; p< _file_read_param.partition_columns.size(); ++p) {
```

Comment:
better to define a local variable `num_part_columns = _file_read_param.partition_columns.size()`